### DISCUSSION TITLE
example of form fill

### DISCUSSION THREAD

**Discussion by kharabela** [qpdf/qpdf#656] [Q&A]
Hey Team,
I want to set pdf fields such as text boxes, dropdowns, checkboxes etc.
Is there a sample for that?

I want to learn -QPDFJob Design usage for the above as well.

**Comment by jberkenbilt** [comment]
QPDFJob doesn't currently have any support for setting form fields since it only supports what you can do from the CLI. From the library, you can set form fields but there are some limitations in the functionality.

[example code](https://github.com/qpdf/qpdf/blob/main/examples/pdf-set-form-values.cc)
[limitations as discussed in the manual](https://qpdf.readthedocs.io/en/stable/cli.html?highlight=appearance#option-generate-appearances)
